lib/SPIRV/SPIRVReader.cpp

```diff
@@ -4462,6 +4467,60 @@ Instruction *SPIRVToLLVM::transOCLBuiltinFromExtInst(SPIRVExtInst *BC,
   return CI;
 }
 
+void SPIRVToLLVM::transAuxDataInst(SPIRVExtInst *BC) {
+  assert(BC->getExtSetKind() == SPIRV::SPIRVEIS_NonSemantic_AuxData);
+  if (!BC->getModule()->preserveAuxData())
```

Comment:
This check can be done at the call-site (outside the loop). Thanks

Reply:
replied to this in other reply, thanks

Reply:
In this case, we will be unnecessarily running the loop if the user did not want to preserve AuxData. It is not PR blocking. So pls feel free to ignore.

Thanks